### ISSUE TITLE
Validator Damping Function

### ISSUE THREAD

**Issue by shawntabrizi** [paritytech/polkadot-sdk#470]
Context: https://kusama.polkassembly.io/post/250

There is a value in the runtime which determines the number of validators the NPoS election algorithm will pick.

We want to use a damping function to control this number between an upper and lower bound based on the distribution of stake/nominations on the network.

TODO:

 - [ ] Sanity check current implementation
 - [ ] Improvements if any to make the algorithm more clean/readable
 - [ ] Configurable pallet flag to disable or enable this feature
 - [ ] Integration Tests
 - [ ] Maybe a little bit of modeling to help outside people understand behavior in diff scenarios

**Comment by rossbulat** [comment]
Would it make sense to close this in favor of #11032?

This issue seems to cover an older approach to dynamically adjusting `ValidatorCount` based on stake/nomination distribution. From a quick look, the current validator-count discussion appears to have moved to #11032, which has more recent context around core needs, approval-voting parameters, and how `desired_targets()` should be dynamic.

If the stake-distribution damping idea is still relevant, maybe it could be reworked around the current staking/election-provider APIs and continued from #11032. Otherwise, this issue may now be obsolete.

**Comment by sigurpol** [comment]
yes I Think it makes sense to close in favor or `11032`. To be noted that 11032 is in a really early /draft stage and needs a lot of discussion / refinement etc. Plus it has dependencies on coretime / RFC17  rework. 
Might be relevant to track also the parallel activity going on Kusama where active set reduction will gradually start after 2.3.0 Kusama runtime grade is enacted - see https://forum.polkadot.network/t/proposal-discussion-adapting-kusama-to-demand-and-its-jam-future/17141/8 + https://hackmd.io/@Lederstrumpf/referendum-kusama-validator-set-reduction-to-700

**Comment by rossbulat** [comment]
Thanks for the clarification, @sigurpol. #11032 is still in an early/draft stage and will need further discussion and refinement. If more specific concerns come up that do not fit cleanly there, dedicated follow-up issues can be opened for those.

The related Kusama validator-set reduction work is also worth tracking as parallel activity, since it targets a similar outcome through a more static mechanism.
